drivers/flash/flash_simulator.c

```diff
@@ -174,9 +174,9 @@ static int flash_range_is_valid(const struct device *dev, off_t offset,
 				size_t len)
 {
 	ARG_UNUSED(dev);
-	if ((offset + len > FLASH_SIMULATOR_FLASH_SIZE +
+	if (((int64_t)(offset + len) > FLASH_SIMULATOR_FLASH_SIZE +
```

Comment:
```suggestion
	if ((int64_t)offset + (int64_t)len >
```
A bit more verbose but avoid integer promotion rules madness entirely.

Reply:
thanks, applied